nilearn/masking.py

```diff
@@ -607,32 +612,44 @@ def compute_brain_mask(target_img, threshold=.5, connected=True,
         Controls the amount of verbosity: higher numbers give
         more messages
 
+    mask_type : {'whole-brain', 'gm', 'wm'}, optional
+        Type of mask to be computed. Default = 'whole-brain'
+
     Returns
     -------
     mask : nibabel.Nifti1Image
         The whole-brain mask (3D image)
     """
     if verbose > 0:
-        print("Template mask computation")
+        msg_to_print = "Template " + mask_type + " mask computation"
+        print(msg_to_print)
 
     target_img = _utils.check_niimg(target_img)
 
-    from .datasets import load_mni152_brain_mask
-    template = load_mni152_brain_mask()
+    if mask_type == 'whole-brain':
+        template = load_mni152_template()
+    elif mask_type == 'gm':
+        template = load_mni152_gm_template()
+    elif mask_type == 'wm':
+        template = load_mni152_wm_template()
```

Comment:
is it possible to get the images  **without** thresholding, eg by providing eg threshold=False ?
This would avoid the awkward threshold -> resample -> threshold sequence.

Reply:
But here we are taking directly the original template. We are not using the loading function, `load_mni152_brain_mask` that takes the template and then resamples+thresholds it.